src/lib/button/button.ts

```diff
@@ -29,50 +29,29 @@ import {
   mixinDisableRipple
 } from '@angular/material/core';
 
-
-// TODO(kara): Convert attribute selectors to classes when attr maps become available
-
 /** Default color palette for round buttons (mat-fab and mat-mini-fab) */
 const DEFAULT_ROUND_BUTTON_COLOR = 'accent';
 
-
 /**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
+ * Map of classes to add to MatButton instances based on host attributes to
+ * style as different variants.
  */
-@Directive({
-  selector: 'button[mat-button], a[mat-button]',
-  host: {'class': 'mat-button'}
-})
-export class MatButtonCssMatStyler {}
-
-/**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
- */
-@Directive({
-  selector: 'button[mat-raised-button], a[mat-raised-button]',
-  host: {'class': 'mat-raised-button'}
-})
-export class MatRaisedButtonCssMatStyler {}
-
-/**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
- */
-@Directive({
-  selector: 'button[mat-icon-button], a[mat-icon-button]',
-  host: {'class': 'mat-icon-button'}
-})
-export class MatIconButtonCssMatStyler {}
+const BUTTON_HOST_ATTRIBUTE_MAP = {
+  '[class.mat-button]': `_hasHostAttributes('mat-button')`,
+  '[class.mat-flat-button]': `_hasHostAttributes('mat-flat-button')`,
+  '[class.mat-icon-button]': `_hasHostAttributes('mat-icon-button')`,
+  '[class.mat-raised-button]': `_hasHostAttributes('mat-raised-button')`,
+  '[class.mat-stroked-button]': `_hasHostAttributes('mat-stroked-button')`,
+  '[class.mat-mini-fab]': `_hasHostAttributes('mat-mini-fab')`,
+  '[class.mat-fab]': `_hasHostAttributes('mat-fab')`,
```

Comment:
It's a bit overkill to add 7 bindings when we know that only one of these is going to apply (the bindings will generate more code and incur higher update costs). We only really need to check once at constructor time and add the class manually.